transport_nantes/photo/views.py

```diff
@@ -85,3 +91,257 @@ def get_context_data(self, **kwargs):
+        photo_sha1 = self.kwargs.get("photo_sha1")
+        photo = get_object_or_404(PhotoEntry, sha1_name=photo_sha1)
+        tn_session_id = self.request.session.get("tn_session")
+        user = self.request.user if self.request.user.is_authenticated else None
+
+        if not photo.accepted:
+            # Author and authorized users can see the photo
+            if self.request.user != photo.user or not self.request.user.has_perm(
+                    'photo.may_see_unaccepted_photos'):
+                return HttpResponseForbidden("Photo not accepted")
+        context["photo"] = photo
+
+        if photo.accepted:
+            self.set_next_and_previous_arrows(context, photo)
+            # Because it's possible that user is not authenticated (anonymous vote)
+            # we need to check if the user has already voted for this photo
+            # using its session id as well.
+            if Vote.objects.filter(
+                    Q(user=user)
+                    | Q(tn_session_id=tn_session_id),
+                    captcha_succeeded=True,
+            ).exists():
+                context["has_voted"] = True
+            else:
+                context["has_voted"] = False
+
+            # To set the proper styles on the vote buttons, we check if the
+            # user has already voted on this photo
+            last_vote = get_user_vote(
+                self.request.user, photo, tn_session_id=tn_session_id)
+
+            print("last_vote : ", last_vote)
+
+            if last_vote:
+                if last_vote.vote_value is True:
+                    context["last_vote"] = "upvote"
+                elif last_vote.vote_value is False:
+                    context["last_vote"] = "downvote"
+
+            if not last_vote and user:
+                # If user is logged in but never voted, we ask for consent
+                # to send them emails
+                context["form"] = SimpleVoteFormWithConsent()
+
+        return context
+
+    def set_next_and_previous_arrows(self, context: dict, photo: 'PhotoEntry') -> None:
+        """
+        Set the next and previous photo arrows
+
+        context is passed by reference.
+        """
+        all_accepted_photos = list(PhotoEntry.objects.filter(accepted=True))
+        current_photo_index = all_accepted_photos.index(photo)
+        if current_photo_index > 0:
+            context["previous_photo"] = all_accepted_photos[
+                current_photo_index - 1]
+        if current_photo_index < len(all_accepted_photos) - 1:
+            context["next_photo"] = all_accepted_photos[
+                current_photo_index + 1]
+
+    def get_initial(self):
+        """Set the initial value in the form"""
+        initial = super().get_initial()
+        initial["photoentry_sha1_name"] = self.kwargs.get("photo_sha1")
+        return initial
+
+    def post(self, request, *args, **kwargs):
+        """
+        Handle POST requests: instantiate a form instance with the passed
```

Comment:
Of course the `post()` function handles POST requests.  Perhaps you could say something more useful here in this important first line, which provides a summary of what the function does?

Reply:
I reworded the docstring to include this information in a clearer way